rep-2015.rst

```diff
@@ -0,0 +1,137 @@
+REP: 2015
+Title: ROS 2 DDS security PKCS#11 URI support
+Author: Miguel Company <miguelcompany@eprosima.com>, Iker Luengo <ikerluengo@eprosima.com>
+Status: Final
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 29-Mar-2023
+Post-History:
+
+Abstract
+========
+
+The DDS-Security specification [1] defines the use of Hardware Security Modules (HSM) and PKCS#11 URIs as an alternative to private keys and certificates stored in the file system.
+Current implementation only supports these tokens to be directly stored in the file system as ``.pem`` files.
+This is a design proposal to support PKCS#11 URIs.
+
+Motivation
+==========
+
+The DDS-Security specification [1] requires the security documents (private keys, certificates, governance and permissions) to be provided to the DDS implementation within the participant's ``properties``.
+It defines a set of reserved properties, that must hold the URI of the corresponding document.
+However, it allows different URI schemes to be used.
```

Comment:
(aside from DDS specification)
as ROS 2 or RMW, i think it would be better to support this feature in general.
in robotics and robot application running on teleoperation or controller devices, those security data would be protected by hardware like TPM(Trusted Platform Module) and Hardware Security Module(HSM).
i think this could be one of the major motivation here.